Sources/DistributedActors/DistributedActors.docc/ClusterSingleton.md

```diff
@@ -0,0 +1,178 @@
+  try await obtainStatus(from: overseer)
+}
+```
+
+Such a proxied call to a distributed actor can take one of three general paths showcased on the diagram below, that we'll explain in depth:
+
+![Diagram showing three nodes, all making calls to the singleton; (a) this node does not know yet where the singleton instance is (b) this node already knows where the singleton is and (c) this node knows that it is hosting the singleton, so the calls are local](cluster_singleton_calls.png)
+
+- **(a)** A call from a freshly joined cluster _member which does not yet know where the singleton is currently hosted_:
+    - Such calls will be buffered by the proxy on the calling side, while it determines the location of the singleton instance.
+    - As the instance is located, the calls are flushed and delivered to the instance on node `"A"`.
+    - If no instance is located within the allocation timeout, or the calls timeout themselves, the calls fail and are dropped from the singleton proxy's buffer.
+- **(b)** A call from a member which has _already discovered where the singleton is hosted_:
+    - Such calls are directly forwarded without any buffering, and only incur a minimal extra local call cost when performing the forwarding to the target instance on `"A"`.
+    - **Note:** This is how all subsequent calls in the (a) scenario are handled, once the singleton has been located and initial messages have been flushed.
+- **(c)** A call on a member which does actually host the singleton instance itself:
+    - Such calls are directly delivered to the local instance.
+    - **Note:** Thanks to location transparency ensured by the `distributed actor` returned by the `singleton.host(...)` call, even if the singleton were to move to another node, the callers from this node don't need to care or change anything -- we still are programming against a location transparent conceptual singleton, even though it happens to be running locally.
+
+### Allocation Strategies
+
+The lifecycle of a cluster singleton is managed by the plugin, and not explicitly by the developer. 
+
+The _allocation_ of a singleton within a cluster consists of actually creating a new _instance_ and _activating_ it - by calling the ``ClusterSingleton/activateSingleton()-6cvj7`` method. Which member of the cluster is to perform this allocation is determined automatically at runtime, based on the state of the cluster, and is determined by the ``ClusterSingletonAllocationStrategySettings`` as configured for the singleton.
+
+The default allocation strategy is ``ClusterSingletonAllocationStrategySettings/byLeadership`` meaning that a singleton is going to be allocated on the _leader_ node of the cluster. 
+
+> Tip: This also means that all singletons in a cluster by default share the same node they run on. This may be sup-optimal in some scenarios, and can be customized by offering a custom allocation strategy, or by future more advanced strategies offered by the library.
```

Comment:
```suggestion
> Tip: This also means that all singletons in a cluster by default share the same node they run on. This may be sup-optimal in some scenarios, and can be customized by providing a custom allocation strategy, or in the future more advanced strategies will be offered by the library.
```